### ISSUE TITLE
Make things that return coordinates return SkyCoord objects somehow

### ISSUE THREAD

**Issue by eteq** [astropy/astroquery#604]
I'm not sure exactly what the scope of this issue might be, but I think it's worth asking at least.  It appears to me that if I do something like querying `Vizier` for catalogs of objects, while I can return their standardized _RAJ2000/DEJ200 columns, there's no straightforward way to get back a `SkyCoord` for the catalog without manually creating one myself.  Has there been any discussion about whether this is desirable?  I can think of some clear ways to do it in e.g. `Vizier` or `SDSS`, at least now that astropy tables support returning tables with columns that are objects... But I'm not sure how general this should be.  Is there someplace that could be plugged into that would allow *all* astroquery tables with appropriate column names or something to get an added skycoord column?  For that matter, is this desirable?

cc @keflavich

**Comment by keflavich** [comment]
This is definitely something I've wanted.  Probably, it should be something configured for each astroquery parser - something that identifies the row names indicating coordinates that are generally returned from a query.  Vizier always has `_RAJ2000`/`_DEJ2000`, SIMBAD will always have something (possibly different?), IPAC will always have something, etc.

What would you suggest for API on this?  `vizier.get_coords(table_result)` or something?

cc @taldcroft is this something we could plug into tables directly, i.e. `tbl.get_coordinates(ra='_RAJ2000', dec='_DEJ2000')`, something like that?

**Comment by eteq** [comment]
You're definitely right that each service would have to specify somehow the right columns (and in some cases might even have to specify particular frames, because they may not always be ICRS).

But my thinking here was more along the lines of embedding the `SkyCoord` *in* the table, rather than having a special query that gives coordinates.  E.g., if I do ``tables = Vizier.get_catalog('something')``, the returned `tables` include an additional *generated* column with a special name like "generated_skycoord" or something.  `Table` already supports that:
```
from astropy.table import Table, Column

t = Table()
t.add_column(Column(name='RA', unit='deg', data=np.random.rand(100)*360))
t.add_column(Column(name='Dec', unit='deg', data=np.random.rand(100)*180-90))
t['generated_skycood'] = SkyCoord(t['RA'], t['Dec'])
```
and then you can just do ``t['generated_skycoord']`` to get out the SkyCoord.

**Comment by keflavich** [comment]
Yes, that sounds great.   It would mean adding another layer, which I think should be a boolean kwarg that defaults to true: `add_skycoord_column`?

**Comment by eteq** [comment]
yep, that sounds perfect.  Although presumably you only want that added for a service that actually returns coordinates, given that they each will need to be customized for whatever name/frame/whatever they return

**Comment by taldcroft** [comment]
I think the conversation already went this way, but I would be hesitant to put any method like `get_coordinate` into the `Table` class.  The functionality of determining (one way or another) which are the relevant columns for creating coordinates seems firmly in the domain of `astroquery` (which knows all about catalogs) rather than `Table`.

**Comment by bsipocz** [comment]
Hmm, I've just wanted to open an issue for this for Simbad... Would be very very nice if we were to return coordinates, quantities, tables when it's reasonable

**Comment by bsipocz** [comment]
(and agree with Tom, that this is purely an astroquery issue, we need to parse the responses into those mixins ourselves here in astroquery)

**Comment by taldcroft** [comment]
And of course it will be a good thing if returning a coordinate mixin instead of plain float columns does not break using Table operations like join, vstack etc. => https://github.com/astropy/astropy/pull/9857.  😼

**Comment by bsipocz** [comment]
oh, I'm certain the astropy PR will be long resolved by the time we get the bottom of this here.

**Comment by ceb8** [comment]
I'll take a stab at this (look out for a WIP PR).

**Comment by jwoillez** [comment]
I find this very interesting. The use case for `(_RAJ2000, _DEJ2000)` is obvious. But what about wanting to use e.g. Gaia (http://vizier.u-strasbg.fr/viz-bin/VizieR-3?-source=I/345/gaia2) to populate a table of `SkyCoord` with parallax and proper motions included, taking into account the 2015.5 epoch? There is also this http://cdsweb.u-strasbg.fr/UCD/ functionality provided by Vizier that gives meaning to columns. They are found in the metadata of each column. They could be used to "magically" perform this conversion to SkyCoord.

**Comment by bsipocz** [comment]
We can start this gradually with the easy, unambigous cases, e.g simbad always returns ra and dec but as float columns while ideally it should be a SkyCoord, that then can feed into other modules.

**Comment by adrn** [comment]
I love this idea!

**Comment by bsipocz** [comment]
@keflavich - This could be a nice project for a student to go through all the modules and make this change, if you have anyone looking for something useful to do?

**Comment by keflavich** [comment]
I will start handing this to students who show interest in coding.

In brief, we're trying to make returns like this:
```python
>>> Simbad.query_objects(['NAME Sgr B2', 'NAME Sgr A*'])
<Table length=2>
  main_id           ra                 dec         coo_err_maj coo_err_min coo_err_angle coo_wavelength     coo_bibcode     user_specified_id object_number_id
                   deg                 deg             mas         mas          deg
   object        float64             float64         float32     float32       int16          str1             object             object           int64
----------- ------------------ ------------------- ----------- ----------- ------------- -------------- ------------------- ----------------- ----------------
NAME Sgr B2 266.83500000000004 -28.385277777777777          --          --            --                                          NAME Sgr B2                1
NAME Sgr A* 266.41681662499997  -29.00782497222222        2.65        1.42             0              R 2011AJ....142...35P       NAME Sgr A*                2
```

return something more like this:
```python
>>> rslt
<Table length=2>
  main_id           ra                 dec         coo_err_maj coo_err_min coo_err_angle coo_wavelength     coo_bibcode     user_specified_id object_number_id                skycoord
                   deg                 deg             mas         mas          deg                                                                                           deg,deg
   object        float64             float64         float32     float32       int16          str1             object             object           int64                      SkyCoord
----------- ------------------ ------------------- ----------- ----------- ------------- -------------- ------------------- ----------------- ---------------- --------------------------------------
NAME Sgr B2 266.83500000000004 -28.385277777777777          --          --            --                                          NAME Sgr B2                1 266.83500000000004,-28.385277777777777
NAME Sgr A* 266.41681662499997  -29.00782497222222        2.65        1.42             0              R 2011AJ....142...35P       NAME Sgr A*                2  266.41681662499997,-29.00782497222222
```

In this specific case, I achieved that by:
```python
rslt['skycoord'] = SkyCoord(rslt['ra'].quantity, rslt['dec'].quantity, frame='fk5')
```
but I didn't check whether the returned coordinates were in fk5 (very important) and it's not obvious how to do this in general (though probably it's not too hard for SIMBAD, Vizier, etc.).

This added column should be _optional_ (specified by keyword, TBD whether it's on or off by default) and should only optionally replace the original columns:
```python
>>> del rslt['ra']
>>> del rslt['dec']
>>> rslt
<Table length=2>
  main_id   coo_err_maj coo_err_min coo_err_angle coo_wavelength     coo_bibcode     user_specified_id object_number_id                skycoord
                mas         mas          deg                                                                                           deg,deg
   object     float32     float32       int16          str1             object             object           int64                      SkyCoord
----------- ----------- ----------- ------------- -------------- ------------------- ----------------- ---------------- --------------------------------------
NAME Sgr B2          --          --            --                                          NAME Sgr B2                1 266.83500000000004,-28.385277777777777
NAME Sgr A*        2.65        1.42             0              R 2011AJ....142...35P       NAME Sgr A*                2  266.41681662499997,-29.00782497222222
```
Why optionally?
 * sometimes there are quirks in the way the data are delivered that matter.  I don't know all the one-off cases, but they'll exist.
 * but we want to allow folks to reduce the size of catalogs on disk when possible

**Comment by ManonMarchand** [comment]
Coordinates in SIMBAD's are always in ICRS and epoch J2000. 

If you do this, could it be off by defaut? My concern is that tables with SkyCoords cannot be written on disk as votables
